### ISSUE TITLE
AppState type inconsistency: "`collaborators.forEach` is not a function"

### ISSUE THREAD

**Issue by wuguishifu** [excalidraw/excalidraw#8637]
The API documentation, built in types, and inspection of the react state all say that the type of `initialData.appState.collaborators` should be an object. However, the code `collaborators.forEach` is also being used, causing a crash when passing in a saved `appState` object to the `initialData` prop on the `<Excalidraw />` element.

Steps to reproduce:

1. Save the app state using the imperative API and see that the collaborators entry is "{}":
  ```tsx
  const appState = excalidrawApi.getAppState();
  console.log(JSON.stringify(appState));
  ```

2. Parse and use this app state in the `initialData` prop:
  ```tsx
  <Excalidraw
    initialData={{
      appState: JSON.parse(appState),
    }}
  />
  ```

This will cause a crash because `collaborators.forEach` is not a function on an object.

**Comment by rajesh2150** [comment]
1.Ensure collaborators is an Array when Restoring:
Modify the code where the saved appState is passed to ensure collaborators is always an array. This can be done by checking its type and converting it if necessary

const parsedAppState = JSON.parse(appState);

// Ensure collaborators is an array
if (!Array.isArray(parsedAppState.collaborators)) {
  parsedAppState.collaborators = [];
}

<Excalidraw
  initialData={{
    appState: parsedAppState,
  }}
/>

2.Adjust Usage of collaborators.forEach

if (collaborators && typeof collaborators === 'object') {
  Object.values(collaborators).forEach(collaborator => {
    // Handle collaborator
  });
}

By Using this above solutions can help.
@wuguishifu 

 Thanks for Your time.

**Comment by wuguishifu** [comment]
True, my solution is to just delete the collaborators entry:

```ts
const parsed = JSON.parse(appState);
if (!Array.isArray(parsed.collaborators)) {
  delete parsed.collaborators;
}
...
```

That being said, this isn't a good solution as it shows an underlying issue with the Excalidraw API.

**Comment by Mrazator** [comment]
Yeah, that's unfortunate. The `collaborators` prop shouldn't really be an array as suggested above, but instance of a `Map`. `JSON.stringify` then turns it (incorrectly) into an object. For now, you can remove it before stringifying or restore it to a Map when parsing.

**Comment by aloniacode** [comment]
Is there no better solution so far?

**Comment by minato32** [comment]
@wuguishifu I'd like to work on this. Root cause (as @Mrazator noted): `collaborators` is a `Map`, which `JSON.stringify` turns into `{}`, so reimporting a saved appState via `initialData` leaves it a plain object and later `collaborators.forEach`/`.has` throws. Fixing by coercing `collaborators` back to a `Map` in `restoreAppState`.
